common/utils.hpp

```diff
@@ -18,10 +18,10 @@ std::string get_abs_path(std::string path) {
     std::filesystem::path current_file_path = std::filesystem::path(__FILE__);
     std::filesystem::path umd_root;
     if (current_file_path.is_absolute()) {
-        umd_root = current_file_path.parent_path().parent_path();
+        umd_root = current_file_path.parent_path().parent_path().parent_path();
```

Comment:
What would be your suggestion on how to resolve this? I don't like having "get_abs_path".

I would imagine we can "install" .yaml files , by adding them explicitly somewhere in CMake config? Then these files would be consumed relatively (to the lib path, if this is even possible? ) just by referencing the name of the file without any dirs, if they end up sitting next to the lib somewhere in /build folder?

I'd prefer if we didn't solve this in this PR, lets leave it for another one. We can put this info in an issue if there's relevant info to be put.

Reply:
A few ideas, depending on how/when/where the tests are invoked.

Typically a unit test is invoked from the build tree without any install step.  Unit tests typically aren't "installed" and shouldn't require that extra step because they you can't just run it form the IDE while developing.

For this type, and if the data is in a single place, then simply setting [WORKING_DIRECTORY](https://cmake.org/cmake/help/latest/command/add_test.html) in CTest would mean running it via CTest or the IDE would have a known `pwd` regardless of where the binary exists.

Another option is to pass the data dir as a #define in the build.  Then the binary can be run even from the CLI and know the path to reach to.

Neither of those solutions allows moving the data to a new root (eg: on a different machine).  If doing that, then you'll want to go through the install step and rearrange files, at that point choose a convention like all tests are together in one dir, and input data is in a subdir named `data`.  Then the code loads from "where am I right now? + data/"

To support running the test locally for development pre-install and also on other machines post-install, then the data lookup might need to try both locations since they differ.